jgrapht-core/src/test/java/org/jgrapht/perf/spanning/MinimumSpanningTreePerformanceTest.java

```diff
@@ -0,0 +1,192 @@
+public class MinimumSpanningTreePerformanceTest
+    extends TestCase
+{
+    private static final int PERF_BENCHMARK_VERTICES_COUNT = 1000;
+    private static final double PERF_BENCHMARK_EDGES_PROP = 0.5;
+    private static final int WARMUP_REPEAT = 5;
+    private static final int REPEAT = 10;
+    private static final long SEED = 13l;
+
+    private static abstract class BenchmarkBase
+    {
+        protected Random rng = new Random(SEED);
+        protected GraphGenerator<Integer, DefaultWeightedEdge, Integer> generator = null;
+        protected DirectedGraph<Integer, DefaultWeightedEdge> graph;
+
+        abstract MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph);
+
+        public void setup()
+        {
+            if (generator == null) {
+                // lazily construct generator
+                generator = new GnpRandomGraphGenerator<>(
+                    PERF_BENCHMARK_VERTICES_COUNT, PERF_BENCHMARK_EDGES_PROP, rng, false);
+            }
+
+            DirectedWeightedPseudograph<Integer, DefaultWeightedEdge> weightedGraph =
+                new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+            this.graph = weightedGraph;
+
+            generator.generateGraph(graph, new IntegerVertexFactory(), null);
+
+            for (DefaultWeightedEdge e : weightedGraph.edgeSet()) {
+                weightedGraph.setEdgeWeight(e, rng.nextDouble());
+            }
+        }
+
+        public void run()
+        {
+            MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> algo = createSolver(graph);
+            algo.getSpanningTree();
+        }
+    }
+
+    public static class PrimBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new PrimMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Prim";
+        }
+    }
+
+    public static class KruskalBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new KruskalMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Kruskal";
+        }
+    }
+
+    public static class BoruvkaBenchmark
+        extends BenchmarkBase
+    {
+        @Override
+        MinimumSpanningTreeAlgorithm<DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph)
+        {
+            return new BoruvkaMinimumSpanningTree<>(graph);
+        }
+
+        @Override
+        public String toString()
+        {
+            return "Boruvka";
+        }
+    }
+
+    public void testBenchmark()
```

Comment:
While the jmh is really nice, it seems to be tailored more towards systems engineering and not algorithms engineering. In the previous benchmarks it felt somehow that I was abusing it to get the functionality that was required. 

With respect to the approximate matching algorithms, I tried to compare the  approximation factor of the algorithms and did not really succeed. They have something called [AuxCounters](http://hg.openjdk.java.net/code-tools/jmh/file/2e94f4ff263b/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_23_AuxCounters.java) but they are really limited, and they even warn against using them in the javadoc (they call it an experimental api, which may be removed in the future, etc.).

Moreover, I seem to be having trouble executing the tests after serious changes (something about some manifest file missing). This is probably my fault, but still the current setup is a bit difficult to use.

The approach that I used in this case, is the standard way of comparing algorithms. Some people like to also test with a cold cache which can be done by setting the `WARMUP_REPEAT` parameter to zero (and of course clear the system's memory buffer cache, using for example /proc/sys/vm/drop_caches in linux).